bigquery/google/cloud/bigquery/client.py

```diff
@@ -1440,6 +1440,97 @@ def query(
 
         return query_job
 
+    def _convert_to_safe_json_dict(self, field, row_value):
```

Comment:
Since these methods don't actually require a client (they don't make any API calls themselves), I'd prefer these live in _helpers.